src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -73,10 +73,44 @@ public OnnxNodeInfo(string name, OnnxShape shape, System.Type type)
         public readonly List<string> InputNames;
         public readonly List<string> OutputNames;
 
+        /// <summary>
+        /// Constructs OnnxModel object from file.
+        /// </summary>
+        /// <param name="modelFile">Model file path</param>
         public OnnxModel(string modelFile)
+            : this (modelFile, -1, false)
+        {
+        }
+
+        /// <summary>
+        /// Constructs OnnxModel object from file.
+        /// </summary>
+        /// <param name="modelFile">Model file path</param>
+        /// <param name="gpuDeviceId">Gpu device ID to execute on. -1 for CPU</param>
+        /// <param name="fallbackToCpu">If true, resumes CPU execution quitely upon GPU error</param>
+        public OnnxModel(string modelFile, int gpuDeviceId = -1, bool fallbackToCpu = false)
         {
             _modelFile = modelFile;
-            _session = new InferenceSession(modelFile);
+
+            if (gpuDeviceId >= 0)
+            {
+                try
+                {
+                    _session = new InferenceSession(modelFile, SessionOptions.MakeSessionOptionWithCudaProvider(gpuDeviceId));
+                }
```

Comment:
is there a way to assert that this is a valid ID before calling on the method?

Reply:
If the ID is invalid, the user gets a fairly descriptive string error message saying "inordinal device ID".

The usage is straightforward... just specify an integer (0, 1, 2 etc.).  MLNet doesn't have a GPU build to put an example for this, otherwise we could introduce a conditional fact to showcase an example.